omp/test/factorization/par_ilu_kernels.cpp

```diff
@@ -159,18 +186,20 @@ TEST_F(ParIlu, KernelComputeParILUIsEquivalentToRef)
     auto coo_omp = Coo::create(omp);
     csr_omp->convert_to(gko::lend(coo_omp));
     initialize_lu(&l_ref, &u_ref, &l_omp, &u_omp);
-    auto u_transpose_ref = u_ref->transpose();
-    auto u_transpose_omp = u_omp->transpose();
+    auto u_transpose_lin_op_ref = u_ref->transpose();
+    auto u_transpose_ptr_ref = static_cast<Csr *>(u_transpose_lin_op_ref.get());
+    auto u_transpose_lin_op_omp = u_omp->transpose();
+    auto u_transpose_ptr_omp = static_cast<Csr *>(u_transpose_lin_op_omp.get());
 
     gko::kernels::reference::par_ilu_factorization::compute_l_u_factors(
         ref, iterations, gko::lend(coo_ref), gko::lend(l_ref),
-        gko::lend(u_ref));
+        u_transpose_ptr_ref);
     gko::kernels::omp::par_ilu_factorization::compute_l_u_factors(
         omp, iterations, gko::lend(coo_omp), gko::lend(l_omp),
-        gko::lend(u_omp));
+        u_transpose_ptr_omp);
 
-    GKO_ASSERT_MTX_NEAR(l_ref, l_omp, .3);
-    GKO_ASSERT_MTX_NEAR(u_ref, u_omp, .3);
+    GKO_ASSERT_MTX_NEAR(l_ref, l_omp, 5e-3);
+    GKO_ASSERT_MTX_NEAR(u_transpose_ptr_ref, u_transpose_ptr_omp, 5e-3);
```

Comment:
If I understood properly we could now increase this by setting iterations to something like `10`. Should we go ahead and do that?

Reply:
We could do that.
I think it might be even better if we add another test that sets the `iterations` to a high number, followed by testing both solutions against each other with a much smaller epsilon.

Reply:
That would be a lot of code duplication though. But I guess your purpose is to test both the iterations == 0 case and one which is extra precise, which would make sense from a unit test POV.

Reply:
Yes, that is pretty much what I would propose. I would actually put this in a separate function now like @hartwiganzt did with the `initialize_lu` and `compute_nnz` (which I realize just now that I should rename it), so we would not have a lot of duplicated code.